megamek/src/megamek/client/bot/princess/MultiTargetFireControl.java

```diff
@@ -0,0 +1,389 @@
+        // units with 'multi-tasker' quirk, multi-gunner vehicles, etc.
+        // are different (and easier to calculate) than optimal firing patterns for other units
+        // because there is no secondary target penalty.
+        // 
+        // So, the basic algorithm is as follows:
+        // For each weapon, calculate the easiest shot. 
+        // Then, solve the backpack problem.
+        
+        List<Mounted> weaponList;
+        
+        if(shooter.usesWeaponBays()) {
+            weaponList = shooter.getWeaponBayList();
+        } else {
+            weaponList = shooter.getWeaponList();
+        }
+        
+        int originalFacing = shooter.getSecondaryFacing();
+        
+        // check all valid secondary facings (turret rotations/torso twists) and arm/flip combination
+        // to see if there's a better firing plan
+        List<Integer> facingChanges = getValidFacingChanges(shooter);
+        facingChanges.add(0); // "no facing change"
+        
+        for(int currentTwist : facingChanges) {
+            shooter.setSecondaryFacing(correctFacing(originalFacing + currentTwist));
+            
+            FiringPlan currentPlan = calculateFiringPlan(shooter, weaponList);
+            currentPlan.setTwist(currentTwist);
+            
+            if(currentPlan.getUtility() > bestPlan.getUtility()) {
+                bestPlan = currentPlan;
+            }
+            
+            // check the plan where the shooter flips its arms
+            if(shooter.canFlipArms()) {
+                shooter.setArmsFlipped(true);
+                
+                currentPlan = calculateFiringPlan(shooter, weaponList);
+                currentPlan.setFlipArms(true);
+                
+                if(currentPlan.getUtility() > bestPlan.getUtility()) {
+                    bestPlan = currentPlan;
+                }
+                
+                // put it back as we found it
+                shooter.setArmsFlipped(false);
+            }
+        }
+        
+        // put it back as we found it
+        shooter.setSecondaryFacing(originalFacing);        
+        
+        return bestPlan;
+    }
+    
+    /**
+     * Get me the best shot that this particular weapon can take. 
+     * @param weapon Weapon to fire.
+     * @return The weapon fire info with the most expected damage. Null if no such thing.
+     */
+    WeaponFireInfo getBestShot(Mounted weapon) {
+        WeaponFireInfo bestShot = null;
+        
+        for(Targetable target : getTargetableEnemyEntities(weapon.getEntity(), owner.getGame(), owner.getFireControlState())) {
+            final int ownerID = (target instanceof Entity) ? ((Entity) target).getOwnerId() : -1;
+            if(owner.getHonorUtil().isEnemyBroken(target.getTargetId(), ownerID, owner.getBehaviorSettings().isForcedWithdrawal())) {
+                owner.log(getClass(), "MultiTargetFireControl.getBestShot", LogLevel.INFO, target.getDisplayName() + " is broken - ignoring");
```

Comment:
Should this be WARN or DEBUG instead of Info?

Reply:
It's "INFO" in the other places where isEnemyBroken is invoked, so I followed that pattern. I could be convinced to call it DEBUG as well.